openfisca_web_api/handlers.py

```diff
@@ -28,10 +25,22 @@ def calculate(tax_benefit_system, input_data):
             entity_result = str(result[entity_index])
         else:
             entity_result = result.tolist()[entity_index]
-
-        dpath.util.new(computation_results, path, entity_result)
-
-    dpath.merge(input_data, computation_results)
+        # Don't use dpath if we have a key that is numeric,
+        # mandatory for dpath upgrade, see https://github.com/dpath-maintainers/dpath-python/issues/160
+        if period.isnumeric():
+            if computation_results == {}:
+                computation_results = {entity_plural: {entity_id: {variable_name: {period: entity_result}}}}
+            else:
+                if entity_plural in computation_results and entity_id in computation_results[entity_plural]:
+                    if variable_name in computation_results[entity_plural][entity_id]:
+                        computation_results[entity_plural][entity_id][variable_name][period] = entity_result
+                    else:
+                        computation_results[entity_plural][entity_id] = {variable_name: {period: entity_result}}
+                else:
+                    computation_results[entity_plural] = {entity_id: {variable_name: {period: entity_result}}}
+        else:
+            dpath.util.new(computation_results, path, entity_result)
```

Comment:
> Memo: dpath.util.new creates the dict entries when they don't exist and updates/extends them if they already exist (see dpath 2.0.5 [`new` documentation](https://github.com/dpath-maintainers/dpath-python/blob/v2.0.5/dpath/util.py#L47-L60)).

Reply:
dpath.util.new removed as I solved a missing case in the _if_ statements.